src/components/PhotoListForm/PhotoListForm.tsx

```diff
@@ -0,0 +1,101 @@
+import Guest from '../Guest/Guest';
+import React, { useState, useEffect } from 'react';
+import Photo from '../Photo/Photo';
+import Checkbox from '../Checkbox/Checkbox';
+
+
+type NewPhoto = {
+  id: number;
+  photoNumber: number;
+  guests: any;
+  description: string;
+}
+
+type WeddingData = {
+  guests: any;
+  changeView: any;
+}
+
+const PhotoListForm: React.FC<WeddingData> = ({guests}) => {
+  const [description, setDescription] = useState('');
+  const [guestsOptions, setGuestsOptions] = useState<any[]>([]);
+  const [photoData, setPhotoData] = useState<any[]>([])
+
+  useEffect(() => {
+    const allGuests = guests.map((guest: any) => {
+        return {...guest, isChecked: false}
+      })
+      setGuestsOptions(allGuests)
+  }, [guests])
+
+  const toggleCheckMark = (guestName: string) => {
+    let toggledList = guestsOptions.map(guest => {
+      if (guest.name === guestName) {
+        guest.isChecked = !guest.isChecked;
+      }
+      return guest
+    })
+    setGuestsOptions(toggledList)
+  }
+
+  const submitPhoto = (event: React.FormEvent) => {
+    event.preventDefault()
+    const guestList = guestsOptions.filter((guest: any) => guest.isChecked)
+    console.log(guestList)
+    const newPhotoData = {
+      id: photoData.length + 1,
+      photoNumber: photoData.length + 1,
+      guests: guestList,
+      description: description
+    }
+    setPhotoData([...photoData, newPhotoData])
```

Comment:
I like the process of toggling the check, mapping the guests, setting them to state and then using that as the data for the filter upon submission

Reply:
Me too! This is a really nice way to handle this, even if it was a little tricky